src/main/java/com/zaxxer/sansorm/internal/AttributeInfo.java

```diff
@@ -0,0 +1,358 @@
+               if (oneToOne != null) {
+                  isOneToOneAnnotated = true;
+               }
+            }
+         }
+      }
+   }
+
+   protected abstract OneToOne extractOneToOneAnnotation();
+
+   protected abstract ManyToOne extractManyToOneAnnotation();
+
+   protected abstract ManyToMany extractManyToManyAnnotation();
+
+   protected abstract OneToMany extractOneToManyAnnotation();
+
+   protected abstract JoinColumns extractJoinColumnsAnnotation();
+
+   protected abstract Transient extractTransientAnnotation();
+
+   protected abstract JoinColumn extractJoinColumnAnnotation();
+
+   protected abstract Enumerated extractEnumeratedAnnotation();
+
+   protected abstract GeneratedValue extractGeneratedValueAnnotation();
+
+   protected abstract Id extractIdAnnotation();
+
+   private void processConvertAnnotation()  {
+      final Convert convertAnnotation = extractConvertAnnotation();
+      if (convertAnnotation != null) {
+         final Class<?> converterClass = convertAnnotation.converter();
+         if (!AttributeConverter.class.isAssignableFrom(converterClass)) {
+            throw new RuntimeException(
+               "Convert annotation only supports converters implementing AttributeConverter");
+         }
+         try {
+            setConverter((AttributeConverter) converterClass.newInstance());
+         }
+         catch (InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+         }
+      }
+   }
+
+   protected abstract Convert extractConvertAnnotation();
+
+   /**
+    * Processes &#64;Column annotated fields.
+    */
+   private void processColumnAnnotation() {
+      final Column columnAnnotation = extractColumnAnnotation();
+      final String columnName = columnAnnotation.name();
+      setColumnName(columnName);
+
+      this.columnTableName = columnAnnotation.table();
+      insertable = columnAnnotation.insertable();
+      updatable = columnAnnotation.updatable();
+   }
+
+   protected abstract Column extractColumnAnnotation();
+
+   protected void processJoinColumnAnnotation() {
+      final JoinColumn joinColumnAnnotation = extractJoinColumnAnnotation();
+      // Is the JoinColumn a self-join?
+      if (type == clazz) {
+         setColumnName(joinColumnAnnotation.name());
+         insertable = joinColumnAnnotation.insertable();
+         updatable = joinColumnAnnotation.updatable();
+      }
+      else {
+         // Do not throw exception. Just ignore. Otherwise classes can not be shared between different JPA implementations.
+//         throw new RuntimeException("JoinColumn annotations can only be self-referencing: " + type.getCanonicalName() + " != "
+//            + clazz.getCanonicalName());
+         System.out.println("WARN: JoinColumn annotations can only be self-referencing and OneToOne. Ignoring " +
+            type.getCanonicalName() + " on "
+            + clazz.getCanonicalName());
+         toBeConsidered = false;
+      }
```

Comment:
I'm not sure I agree with not throwing here.  It is a non-goal of SansOrm to allow "transparent" migration from objects annotated for a full ORM, for example Hibernate, to SansOrm.

I feel it would be better to clearly fail, such that users know what is definitely not supported by SansOrm.

While it may be possible to go from SanOrm -> "Full ORM", with respect to annotations, it is defintely not possible to go the other direction -- and is not a goal to do so as it is simply not possible unless SansOrm were in fact a full ORM.

Reply:
Note that that is *not* true of annotations that SansOrm does not support at all.  For example, ``@OneToMany`` may simply be ignored.  However, ``@JoinColumn`` *is* a *partially`` supported annotation, so we should throw when unsupported uses are discovered.

Reply:
GetterAnnotatedPitMainEntity shows an automatically with IntelliJ generated entity. We could not use it as-is with SansOrm because SansOrm will throw an Exception when processing
```
   @ManyToOne
   @JoinColumn(name = "PIT_IDENT", referencedColumnName = "PIR_PIT_IDENT", nullable = false)
   public GetterAnnotatedPitReferenceEntity getPitReferenceByPitIdent() {
      return pitReferenceByPitIdent;
   }
```